src/psyclone/tests/gocean1p0_transformations_test.py

```diff
@@ -5,18 +5,20 @@
 # whose members are identified at https://puma.nerc.ac.uk/trac/GungHo/wiki
```

Comment:
Sorry to be a pain but I've just realised that the `LoopSwapTrans` is a general transformation but we don't have any tests for it with any other API. In fact, for the dynamo0p3 API I think it is not applicable since the only time we have nested loops there is when a loop over cells is coloured. We only do that if we're parallelising with OpenMP but I think the attempt to do that would fail once the loops are swapped (because we check that any loop over colours is performed sequentially). I think therefore that we have three choices: alter `SwapLoopTransform` to check that the API=="GOcean", rename it to e.g. `GOSwapLoopTransform` or add a test for Dynamo0p3 with the proviso that it's not actually useful at the moment. My preference would be for the latter because, although it's not useful right now, it is clearly a general concept and should be treated as such.

Reply:
I had a look and I can't even find a (valid) example where I could swap loops in dynamo0p3 - as you said the only nested loops are coloured OMP loops, which can't be swapped (and while I didn't check, I wouldn't be surprised that the OMP statement in the schedule will make the swap transform to abort). So I am not even sure how to create a valid/useful testcase for that. I would suggest to do the first two: check the API, and rename the transform *and those transforms really should be split into different classes: one file with the base transforms, and then one for each API ... but you know that probably ;) ).

Reply:
OK, let's do the first two. (And yes, you're right about splitting the file but that's something for another issue :-) )

Reply:
Done - I couldn't find an easy way to test the API (since a transform doesn't easily have access to it), but I tested the loop type to make sure it's GOLoop, which should be sufficient.
I noticed that three gocean transforms in transformations.py are called GOceanXXX, not GOXXX - but I followed your suggestion to use GOSwap...